servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/RoundRobinToDefaultLBMigrationProvider.java

```diff
@@ -0,0 +1,154 @@
+        private Duration healthCheckInterval = DEFAULT_HEALTH_CHECK_INTERVAL;
+        private Duration healthCheckJitter = DEFAULT_HEALTH_CHECK_JITTER;
+        private int healthCheckFailedConnectionsThreshold = DEFAULT_HEALTH_CHECK_FAILED_CONNECTIONS_THRESHOLD;
+        private Duration healthCheckResubscribeInterval = DEFAULT_HEALTH_CHECK_RESUBSCRIBE_INTERVAL;
+        private Duration healthCheckResubscribeJitter = DEFAULT_HEALTH_CHECK_JITTER;
+
+        DefaultLoadBalancerRoundRobinBuilder(final String id) {
+            this.id = id;
+        }
+
+        @Override
+        public RoundRobinLoadBalancerBuilder<ResolvedAddress, C> linearSearchSpace(int linearSearchSpace) {
+            this.linearSearchSpace = ensureNonNegative(linearSearchSpace, "linearSearchSpace");
+            return this;
+        }
+
+        @Override
+        public RoundRobinLoadBalancerBuilder<ResolvedAddress, C> backgroundExecutor(Executor backgroundExecutor) {
+            this.backgroundExecutor = backgroundExecutor;
+            return this;
+        }
+
+        @Override
+        public RoundRobinLoadBalancerBuilder<ResolvedAddress, C> healthCheckInterval(
+                Duration interval, Duration jitter) {
+            validateHealthCheckIntervals(interval, jitter);
+            this.healthCheckInterval = interval;
+            this.healthCheckJitter = jitter;
+            return this;
+        }
+
+        @Override
+        public RoundRobinLoadBalancerBuilder<ResolvedAddress, C> healthCheckResubscribeInterval(
+                Duration interval, Duration jitter) {
+            validateHealthCheckIntervals(interval, jitter);
+            this.healthCheckResubscribeInterval = interval;
+            this.healthCheckResubscribeJitter = jitter;
+            return this;
+        }
+
+        @Override
+        public RoundRobinLoadBalancerBuilder<ResolvedAddress, C> healthCheckFailedConnectionsThreshold(int threshold) {
+            if (threshold == 0) {
+                throw new IllegalArgumentException("Health check failed connections threshold should not be 0");
+            }
+            this.healthCheckFailedConnectionsThreshold = threshold;
+            return this;
+        }
+
+        @Override
+        public LoadBalancerFactory<ResolvedAddress, C> build() {
+            OutlierDetectorConfig outlierDetectorConfig = new OutlierDetectorConfig.Builder()
+                    .enforcingFailurePercentage(0)
+                    .enforcingSuccessRate(0)
+                    .enforcingConsecutive5xx(0)
+                    .failedConnectionsThreshold(healthCheckFailedConnectionsThreshold)
+                    .failureDetectorInterval(healthCheckInterval, healthCheckJitter)
+                    .serviceDiscoveryResubscribeInterval(healthCheckResubscribeInterval, healthCheckResubscribeJitter)
+                    .build();
+
+            LoadBalancingPolicy<ResolvedAddress, C> loadBalancingPolicy = new RoundRobinLoadBalancingPolicy.Builder()
```

Comment:
Another side note: to help with discoverability of different policies we can consider adding smth like `LoadBalancingPolicies` class with static factories